teuthology/suite/run.py

```diff
@@ -189,30 +194,42 @@ def choose_teuthology_branch(self):
         log.info("teuthology branch: %s", teuthology_branch)
         return teuthology_branch
 
+    @property
+    def suite_repo_name(self):
+        if self.args.suite_repo:
+            return self.args.suite_repo.split('/')[-1].rstrip('.git')
+        else:
+            return 'ceph-qa-suite'
+
     def choose_suite_branch(self):
+        suite_repo_name = self.suite_repo_name
         suite_branch = self.args.suite_branch
         ceph_branch = self.args.ceph_branch
         if suite_branch and suite_branch != 'master':
-            if not util.git_branch_exists('ceph-qa-suite', suite_branch):
-                exc = BranchNotFoundError(suite_branch, 'ceph-qa-suite.git')
+            if not util.git_branch_exists(suite_repo_name, suite_branch):
```

Comment:
Ah, it will but it doesn't have to be that way of course. I'll work on that in the morning